_posts/2021-03-10-dask_memory_usage.md

```diff
@@ -0,0 +1,261 @@
+    cluster = LocalCluster(n_workers=4, threads_per_worker=1,
+                           memory_limit=None)
+    # Install dask-memusage:
+    dask_memusage.install(cluster.scheduler, "memusage.csv")
+    client = Client(cluster)
+
+    # Create the task graph:
+    files = from_sequence(Path(directory).iterdir())
+    graph = files.map(calculate_top_10)
+    graph.visualize(filename="example2.png", rankdir="TD")
+
+    # Run the calculations:
+    for result in graph.compute():
+        print(result)
+    # ... do something with results ...
+
+
+if __name__ == '__main__':
+    main(sys.argv[1])
+```
+
+Here's what the task graph looks like:
+
+<img src="/images/dask_memusage/example2.png" style="width: 75%; margin: 2em;">
+
+Plenty of parallelism!
+
+We can run the program on some files:
+
+```shell-session
+$ pip install dask[bag] dask_memusage
+$ python example2.py files/
+('frankenstein.txt', [('that', 1016), ('was', 1021), ('in', 1180), ('a', 1438), ('my', 1751), ('to', 2164), ('i', 2754), ('of', 2761), ('and', 3025), ('the', 4339)])
+('pride_and_prejudice.txt', [('she', 1660), ('i', 1730), ('was', 1832), ('in', 1904), ('a', 1981), ('her', 2142), ('and', 3503), ('of', 3705), ('to', 4188), ('the', 4492)])
+('greatgatsby.txt', [('that', 564), ('was', 760), ('he', 770), ('in', 849), ('i', 999), ('to', 1197), ('of', 1224), ('a', 1440), ('and', 1565), ('the', 2543)])
+('big.txt', [('his', 40032), ('was', 45356), ('that', 47924), ('he', 48276), ('a', 83228), ('in', 86832), ('to', 114184), ('and', 152284), ('of', 159888), ('the', 314908)])
+```
+
+As one would expect, the most common words are stem words, but there is still some variation in order.
+
+Next, let's look at the results from `dask-memusage`.
+
+## `dask-memusage` output, and how it works
+
+You'll notice that the actual use of `dask-memusage` involves just one extra line, other than the import:
+
+```python
+dask_memusage.install(cluster.scheduler, "memusage.csv")
+```
+
+What this will do is poll the process at intervals for peak memory usage, broken down by task.
+In this case, here's what `memusage.csv` looks like:
+
+```csv
+task_key,min_memory_mb,max_memory_mb
+"('from_sequence-3637e6ff937ef8488894df60a80f62ed', 3)",51.2421875,51.2421875
+"('from_sequence-3637e6ff937ef8488894df60a80f62ed', 0)",51.70703125,51.70703125
+"('from_sequence-3637e6ff937ef8488894df60a80f62ed', 1)",51.28125,51.78515625
+"('from_sequence-3637e6ff937ef8488894df60a80f62ed', 2)",51.30859375,51.30859375
+"('calculate_top_10-afc867e38c3bd0aac8c18bb00d3634ca', 2)",56.19140625,56.19140625
+"('calculate_top_10-afc867e38c3bd0aac8c18bb00d3634ca', 0)",51.70703125,54.26953125
+"('calculate_top_10-afc867e38c3bd0aac8c18bb00d3634ca', 1)",52.30078125,52.30078125
+"('calculate_top_10-afc867e38c3bd0aac8c18bb00d3634ca', 3)",51.48046875,384.00390625
+```
+
+For each task in the graph we are told minimum memory usage and peak memory usage, in MB.
+
+In more readable form:
```

Comment:
The table rendered from this markdown [here](https://github.com/itamarst/dask-blog/blob/gh-pages/_posts/2021-03-10-dask_memory_usage.md) isn't all that readable. 

I think you have to have three or more hyphens used to create a heading for each column, like this:

```
| Syntax      | Description |
| ----------- | ----------- |
| Header      | Title       |
| Paragraph   | Text        |
```

which renders like this:

| Syntax      | Description |
| ----------- | ----------- |
| Header      | Title       |
| Paragraph   | Text        |

Reply:
So I previewed this in Jekyll, and it looked fine there, so this may be limitation of GitHub renderer. But bold titles will be nicer, yes.